### ISSUE TITLE
`max-statements-per-line`: allow specific statements

### ISSUE THREAD

**Issue by Cherry** [eslint-stylistic/eslint-stylistic#211]
### Clear and concise description of the problem

It would great if this rule could ignore specific statements, such as `continue` and `break`, since these are often used in loops and it's a common stylistic choice from my experience to do these on a single line.

For example, a contrived loop. There's much better ways to do this in modern JS, but illustrates the request:
```javascript

const items = [
	{
		include: true,
		value: 1,
	},
	{
		include: false,
		value: 2,
	},
	{
		include: true,
		value: 3,
	},
];

for (const item of items) {
	if (!item.include) { continue; } // errors with @stylistic/max-statements-per-line: ["error", { "max": 1 }]
	console.log(item.value);
	// do something with item
}

```

While I do want to prevent things like:
```javascript

if (condition) { bar = 1; } else { bar = 2; }

// or

for (var i = 0; i < length; ++i) { bar = 1; }
```

I think allowing `break` or `continue` as exceptions to this could be worthwhile

### Suggested solution

Perhaps something like this could be a useful configuration?

```js
'@stylistic/max-statements-per-line': [
		'error',
		{
			max: 1,
			ignoredStatements: [
				'ContinueStatement',
				'BreakStatement',
			],
		},
	],
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by Cherry** [comment]
Hmm yeah that definitely feels odd. I wouldn't have expected the wrapping in braces to produce different results 🤔

The primary difference in the AST appears to be that the first has the `ContinueStatement` as the `consequent`, whereas the second has a `BlockStatement` with the `ContinueStatement` in the `body`.
![](https://up.jross.me/byw9owcl)
https://astexplorer.net/#/gist/eac93feb074c59397c23955caa182f84/b62518fdd7cbc027c195b2496617b3ea0e068b16

**Comment by 500-internal-server-error** [comment]
+1, would love the additional flexibility - e.g., sometimes I like to do `case X: assignvar = 3; break;` or `case Y: callfunc(3); break;` which unfortunately counts as more than 1 statement, which while I generally do prefer having, sometimes things are short enough for me to decide it's ok.

```js
switch (lorem) {
    case ipsum: dolor(); break;
    case sit: amet(); break;
    // a long list of cases with a similar format in the body
}
```

A bit of a longer shot would be multi-statements but still short, which ideally I'd like to be able to exempt as well but I'm not too / as bothered about.

```js
var x;
switch (lorem) {
    case ipsum: x = 3; dolor(); /* could be many statements but still "short" */ break;
    case sit: x = 4; amet(); break;
    // a long list of cases, with many short statements
}

// this should still be disallowed
// if (foo) { bar = 3; foobar(); }
// forcing to expand 1 if statement is fine, forcing to expand
// the case into a block is a waste of space when there are many cases
```
